improver_tests/cli/test_init.py

```diff
@@ -165,113 +165,109 @@ def test_with_output(self, m):
+    These cubes should be the same as the setup cubes.
 
     Returns:
-        iris.cube.Cube:
-            The extracted cube.
-
-    Raises:
-        ValueError:
-            Error to replicate the behaviour of the call of maybe_coerce_with,
-            where loading a cube with an invalid constraint results in a
-            ValueError.
+        iris.cube.CubeList:
+            The CubeList.
     """
-    del func
-    if constraints:
-        cube = cube.extract(constraints)
-    if isinstance(cube, CubeList):
-        (cube,) = cube.copy()
-    if cube is None:
-        raise ValueError
-    return cube
+    return CubeList(
+        [
+            set_up_variable_cube(
+                data=np.zeros((2, 2), dtype=np.float32), name="wind_speed"
+            ),
+            set_up_variable_cube(
+                data=np.zeros((2, 2), dtype=np.float32), name="wind_from_direction"
+            ),
+        ]
+    )
 
 
 class Test_create_constrained_inputcubelist_converter(unittest.TestCase):
-    """Tests the creature constraint_inputcubelist_converter"""
+    """Tests the create_constraint_inputcubelist_converter"""
 
     def setUp(self):
         data = np.zeros((2, 2), dtype=np.float32)
         self.wind_speed_cube = set_up_variable_cube(data, name="wind_speed")
         self.wind_dir_cube = set_up_variable_cube(data, name="wind_from_direction")
         self.wind_cubes = CubeList([self.wind_speed_cube, self.wind_dir_cube])
 
-    @patch("improver.cli.maybe_coerce_with", return_value="return")
-    def test_basic(self, m):
-        """Tests that it returns a function which itself returns 2 cubes"""
-        result = create_constrained_inputcubelist_converter(
-            "wind_speed", "wind_from_direction"
+    def test_basic(self):
+        """Tests that providing no valid constraints raises a ConstraintMismatchError."""
+        func = create_constrained_inputcubelist_converter(
+            lambda cube: cube.name() in ["wind_speed", "grid_eastward_wind"]
         )
-        result("foo")
-        m.assert_any_call(load_cube, "foo", constraints="wind_speed")
-        m.assert_any_call(load_cube, "foo", constraints="wind_from_direction")
-        self.assertEqual(m.call_count, 2)
-
-    @patch("improver.cli.maybe_coerce_with", return_value="return")
-    def test_list(self, m):
-        """Tests that a list returns a function which itself returns 2 cubes"""
-        result = create_constrained_inputcubelist_converter(
-            ["wind_speed"], ["wind_from_direction"]
+        result = func(self.wind_cubes)
+        self.assertIn(self.wind_speed_cube, result)
+        self.assertEqual(1, len(result))
+
+    def test_two_lists(self):
+        """Tests a basic creation of create_constrained_inputcubelist_converter"""
+        func = create_constrained_inputcubelist_converter(
+            lambda cube: cube.name() in ["wind_speed"],
+            lambda cube: cube.name() in ["wind_from_direction"],
```

Comment:
Following Tomek's review, I presume that a few of these unit tests could also be simplified to e.g.
```
func = create_constrained_inputcubelist_converter(
    "wind_speed", "wind_from_direction"
)
```
The unit tests that I think need updating are: `test_two_lists`, `test_basic_given_str`, `test_list_two_diff_shapes` and `test_list_no_match`, as these tests have only one item within the list within the lambda functions.